### PR TITLE
prep record struct for layout

### PR DESCRIPTION
**Pull request by TwoClocks**
Add `TypeLayout` to `Record` and `FieldLayout` to `Field` in `Type.zig`

`TypeLayout` is probably too fat. I think I can cut it in half, but I'd like to finish msvc and MinGW layout emulation before doing that. It helps w/ diff-ing the rust code.

Removed `size` and `alignment` from `Record` as they are duplicative of `TypeLayout`

Changed `bits` in `Field` to be nullable, as 0 and unset mean different things to the layout code.

### CODE REVIEW

File: src/Type.zig [hunk 1]
```diff
@@ -162,24 +162,70 @@ pub const Enum = struct {
...
 };
+// might not need all 4 of these when finished,
+// but currently it helps havaing all 4 when diff-ing
+// the reust code.
+pub const TypeLayout = struct {
+    /// The size of the type in bits.
+    ///
+    /// This is the value returned by `sizeof` and C and `std::mem::size_of` in Rust
+    /// (but in bits instead of bytes). This is a multiple of `pointer_alignment_bits`.
+    size_bits: u29,
```
Comment (Vexu) [inline src/Type.zig:173]: It's weird that this uses `u29` when we allow array sizes up to `u64`. How does the struct layout code handle huge arrays in structs? How do GCC/clang handle it?

**Comment by ehaas** [comment]
All of the size fields in the Rust code (in FieldLayout and TypeLayout) are u64 and not u29

File: src/Type.zig (hunk 1 above)
Comment (TwoClocks) [reply src/Type.zig:173]: That's not wierd, it's wrong. Fixed.
Thanks!

**Comment by TwoClocks** [comment]
Yep, they should all be u64.
Thanks

File: src/Type.zig [hunk 2]
```diff
@@ -162,24 +162,70 @@ pub const Enum = struct {
...
+    ///
+    /// This is the value returned by `std::mem::align_of` in Rust
+    /// (but in bits instead of bytes). `size_bits` is a multiple of this value.
+    pointer_alignment_bits: u64,
+    /// The required alignment of the type in bits.
+    ///
+    /// This value is only used by MSVC targets. It is 8 on all other
+    /// targets. On MSVC targets, this value restricts the effects of `#pragma pack` except
+    /// in some cases involving bit-fields.
+    required_alignmnet_bits: u64,
```
Comment (ehaas) [inline src/Type.zig:189]: typo/spelling error - should be `required_alignment_bits`

Comment (TwoClocks) [reply src/Type.zig]: I am so dyslexic, I had to stare at your comment for 2min before I saw the difference :smile: 
I need to get neovim to spell check snake & camelCase.

**Pull request merged**